### PR TITLE
Improvements for relationship view processing in app inspector

### PR DESCRIPTION
**Pull request by algairim**
Signed-off-by: Mykola Mandra <mykola.mandra@cloudsoftcorp.com>

### CODE REVIEW

File: ui-modules/app-inspector/app/components/entity-tree/entity-tree.directive.js [hunk 1]
```diff
@@ -141,52 +134,60 @@ export function entityTreeDirective() {
...
-                                        // 4.2. Move 'hosted_on' entity to a new 'host_for' parent.
-                                        moveEntityToParent(relatedEntity, entity, entities, VIEW_HOST_FOR_HOSTED_ON);
+            function addOtherViews(entities, relationships) {
+                let otherViews = Array.from($scope.viewModes).filter(r => r !== VIEW_PARENT_CHILD);
+                otherViews.forEach(view => {
+
+                    // Define parent and child identifiers
+                    const otherParentChildIdentifiers = view.split(RELATIONSHIP_VIEW_DELIMITER);
+                    const OTHER_PARENT = otherParentChildIdentifiers[0];
+                    const OTHER_CHILD = otherParentChildIdentifiers[1];
```
Comment (algairim) [inline ui-modules/app-inspector/app/components/entity-tree/entity-tree.directive.js:150]: This is the new code.

File: ui-modules/app-inspector/app/components/entity-tree/entity-tree.directive.js [hunk 2]
```diff
@@ -141,55 +134,106 @@ export function entityTreeDirective() {
...
-                                        flipParentAndChild(relatedEntity, entity, entities, VIEW_HOST_FOR_HOSTED_ON);
-                                    } else {
-                                        // 4.2. Move 'hosted_on' entity to a new 'host_for' parent.
-                                        moveEntityToParent(relatedEntity, entity, entities, VIEW_HOST_FOR_HOSTED_ON);
+                            // Label every 'OTHER_PARENT' entity to display and highlight in 'OTHER_PARENT/OTHER_CHILD'
+                            // view mode, only if it does not play the role of 'OTHER_CHILD' at the same time.
+                            if (!childRole) {
+                                displayEntityInView(entity, view);
+                                highlightEntityInView(entity, view);
+                            }
```
Comment (algairim) [inline ui-modules/app-inspector/app/components/entity-tree/entity-tree.directive.js:164]: This is the new bit in the code block refactored above.

File: ui-modules/app-inspector/app/components/entity-tree/entity-tree.directive.js [hunk 3]
```diff
@@ -141,55 +134,106 @@ export function entityTreeDirective() {
...
-                        displayEntityInView(entity, VIEW_HOST_FOR_HOSTED_ON);
-                    }
+                            // Display original position for any other entity under 'OTHER_PARENT/OTHER_CHILD' view. Do
+                            // no highlight entities that are required to be displayed but do not belong to this view.
+                            displayEntityInView(entity, view);
+                        }
+                    });
+
+                    // Phase 2. Look through all entities found again and process entities that play both roles:
+                    //          'OTHER_PARENT' and 'OTHER_CHILD'.
```
Comment (algairim) [inline ui-modules/app-inspector/app/components/entity-tree/entity-tree.directive.js:196]: New phase to cover edge case when entity plays role of parent and child in the view.

**Comment by ahgittin** [comment]
looks good

**Pull request merged**